packages/compiler/lib/decorators.ts

```diff
@@ -87,6 +91,7 @@ const docsKey = createStateSymbol("docs");
  * @doc can be specified on any language element -- a model, an operation, a namespace, etc.
  */
 export function $doc(context: DecoratorContext, target: Type, text: string, sourceObject?: Type) {
+  validateDecoratorUniqueOnNode(context, target, $doc);
```

Comment:
Feels like most decorators kind would want this rule. Not having it is more of the exception. `@extension` for openapi for examplke that you might want to call multiple times with different values.

Wondering if we can have something more built-in the compiler then

Reply:
Yeah, not sure how we'd express it. It reminds me of .NET AttributeUsageAttribute, but then we don't have decorators on decorators. Design issue?